backend_tv/adjust_src.cpp

```diff
@@ -481,4 +481,43 @@ void fixupOptimizedTgt(llvm::Function *tgt) {
     myAllocCall->eraseFromParent();
 }
 
+/**
+ * Given an intToPtr instruction, checks for a round trip 
+ * from a ptrToInt instruction, then replaces with a single GEP instruction.
+ * This translation can be verified with physical pointers.
+ */
+void tryReplaceRoundTrip(llvm::IntToPtrInst *intToPtr) {
+  if (!intToPtr)
```

Comment:
you don't need to do this check if you've already tested this pointer for non-null elsewhere (as you have done). if you don't feel comfortable totally removing this check, you can replace it with an assertion.